devtools/db_import/db_import/db.py

```diff
@@ -47,3 +47,38 @@ def insert_rows(self, table: Union[str, Table], rows: Iterable[Mapping[str,
 
   def get_table(self, table: Union[str, Table]) -> Table:
     ...
+
+
+def query_returns_non_empty_result(
+    client: Client,
+    sql: str,
+    parameters: dict[str, str],
+) -> bool:
+  job_config = bigquery.QueryJobConfig(query_parameters=[
+      bigquery.ScalarQueryParameter(key, "STRING", value)
+      for key, value in parameters.items()
+  ])
+
+  rows = list(
+      client.query(
+          f"SELECT COUNT(*) as count FROM ({sql})",
```

Comment:
Will this result in SQL injection? How do we ensure the `sql` is trustable

Reply:
We ensure that `sql` is trustable the same way we ensure that the sql argument to `client.query` is trustable.

I see this function as a convenience wrapper around `client.query` - which also takes an arbitrary SQL string as input, hence there is no bigger risk than using `client.query` directly, right?

I could potentially remove the count aggregation here and make it the caller's responsibility to return a `cound` field right away. Then I could just forward the sql untouched. Would that make more sense to you?

Reply:
Another option is to hand off this task to a SQL parser library like sqlglot.

The following expression should do the same:

```python
sqlglot.subquery(sqlglot.parse_one(sql)).select("COUNT(*) as count").sql()
```

sqlglot parses the subquery into an AST and reemits it in the `.sql()` call. This should result in something that fails whenever `sql` is not a valid select expression (modulo sqlglot bugs of course)

WDYT about that?

Reply:
Makes sense. I also see `bigquery.QueryJobConfig` is used to pass parameters from untrusted sources. The current approach is reasonable to me.

Reply:
SG. Leving as-is.